Repair package spec for legacy csproj

drewgillies opened the pull request:
Package spec generated by legacy csproj had some gaps for supporting UWP restores.
@emgarten

drewgillies committed:
Repair package spec for legacy csproj

drewgillies commented:
Note that it would be good to add some unit tests for this--but I'd like a build for running scenarios tonight, so I'll get onto the unit test coverage next.

emgarten commented on src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs:
```diff
@@ -278,26 +281,35 @@ private PackageSpec GetPackageSpec()
                 RestoreMetadata = new ProjectRestoreMetadata
                 {
-                    OutputType = RestoreOutputType.NETCore,
+                    OutputType = RestoreOutputType.UAP,
```
This should stay as NETCore, only the UWPProjectJson should use this type from how I understand the goal.

emgarten commented on src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs:
```diff
@@ -278,26 +281,35 @@ private PackageSpec GetPackageSpec()
+            return new PackageSpec()
             {
                 Name = _projectName ?? _projectUniqueName,
+                Version = new NuGetVersion(1, 0, 0, 0, (string)null, string.Empty),
```
This should read the Version property from the project file. ProjectVersion may override it. This all needs to match pack behavior.

drewgillies replied:
Discussed offline--will circle back to this.

emgarten commented on src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs:
```diff
@@ -278,26 +281,35 @@ private PackageSpec GetPackageSpec()
+                TargetFrameworks = tfis,
+                ContentFiles = new string[] { },
+                Dependencies = packageReferences,
+                IsDefaultVersion = true,
```
I would skip setting this

emgarten commented on src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs:
```diff
@@ -278,26 +281,35 @@ private PackageSpec GetPackageSpec()
                     ProjectUniqueName = _projectFullPath,
                     OriginalTargetFrameworks = tfis
-                        .Select(tfi => tfi.FrameworkName.GetShortFolderName())
+                        .Select(tfi => tfi.FrameworkName?.GetShortFolderName())
```
This should never be null, right?
Also, if it were null you would need to filters those out

emgarten reviewed (changes requested):
LegacyPackageRef should be NETCore output type. UAP is just for project.json.

drewgillies commented:
Some rebasing here so the iteration won't be obvious, @emgarten. however, it's just the addressing of your feedback (small delta) and the reverting of the UAP test and spec validation changes, as these are no longer needed after changing the output type.

emgarten commented on src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs:
```diff
@@ -253,7 +255,7 @@ private PackageSpec GetPackageSpec()
-                .Select(ToPackageLibraryDependency);
+                .Select(ToPackageLibraryDependency).ToList();
 
             var packageTargetFallback = _project.PackageTargetFallback?.Split(new[] { ';' })
```
This should use https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs#L405

That can be fixed as part of the trimming issue later if needed

emgarten reviewed (approved):
LGTM
